### PR TITLE
*: Add operation of `admin show ddl jobs`

### LINKED ISSUE #4311: Add a interface to get DDL jobs from job queue
Currently there is no way to view the waiting DDL jobs, which is very inconvenient for debugging.

I propose to add SQL syntax like 'admin show ddl-jobs' to solve this.

### PR DESCRIPTION
**Pull request by zimulala**
Add a command to show the DDL jobs in the waiting queue and history queue.
Fix #4311

### CODE REVIEW

**Comment by zimulala** [comment]
PTAL @winkyao @tiancaiamao

File: inspectkv/inspectkv.go [hunk 1]
```diff
@@ -64,6 +64,43 @@ func GetDDLInfo(txn kv.Transaction) (*DDLInfo, error) {
...
+func GetDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
+	t := meta.NewMeta(txn)
+	cnt, err := t.DDLJobQueueLen()
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+
+	jobs := make([]*model.Job, cnt)
+	for i := range jobs {
+		jobs[i], err = t.GetDDLJob(int64(i))
```
Comment (tiancaiamao) [inline inspectkv/inspectkv.go]: Is DDLJobQueueLen and GetDDLJob in an atomic operation or snapshot?
Will DDLJobQueueLen change during this line?

File: executor/builder.go [hunk 2]
```diff
@@ -169,6 +171,29 @@ func (b *executorBuilder) buildShowDDL(v *plan.ShowDDL) Executor {
...
+		return nil
+	}
+	e.cnt = len(e.jobs)
+	e.historyJobs, err = inspectkv.GetHistoryDDLJobs(e.ctx.Txn())
+	if err != nil {
+		b.err = errors.Trace(err)
+		return nil
+	}
+	if e.cnt < len(e.historyJobs) {
+		e.cnt = len(e.historyJobs)
```
Comment (tiancaiamao) [inline executor/builder.go]: cnt means finished jobs count?

**Comment by winkyao** [comment]
maybe we can use `admin show ddl jobs` to show queued ddj jobs and `admin show ddl history jobs` to show the history ddl jobs? I think it is more graceful.

File: inspectkv/inspectkv.go (hunk 1 above)
Comment (zimulala) [inline inspectkv/inspectkv.go]: This function uses a transaction.

File: executor/builder.go (hunk 2 above)
Comment (zimulala) [inline executor/builder.go]: Yep, I use the longer jops.

**Comment by zimulala** [comment]
@winkyao It will do in the next PR. I will add a `TODO`.

File: inspectkv/inspectkv.go [hunk 3]
```diff
@@ -64,6 +64,43 @@ func GetDDLInfo(txn kv.Transaction) (*DDLInfo, error) {
 	return info, nil
 }
 
+// GetDDLJobs returns the DDL jobs and an error.
+func GetDDLJobs(txn kv.Transaction) ([]*model.Job, error) {
```
Comment (hanfei1991) [inline inspectkv/inspectkv.go]: Why put these two functions in `inspectkv.go`

Comment (zimulala) [reply inspectkv/inspectkv.go]: Because `GetDDLInfo` and some admin functions all in this file. @hanfei1991

**Comment by coocood** [comment]
@zimulala 
I think we better separate history and current job in this PR.
A job is either current or history, it doesn't make sense for a row with both current job column and history job column.

**Comment by zimulala** [comment]
There are all jobs. And I think it would be more convenient to use a statement when letting customers help check DDL jobs information. @coocood

**Comment by coocood** [comment]
Or use a column called `State` with value `waiting/running/finished/canceled`?
@zimulala

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by zimulala** [comment]
PTAL @coocood

**Comment by coocood** [comment]
LGTM

**Pull request merged**